packages/cmf/src/cmfConnect.js

```diff
@@ -272,14 +311,63 @@ export default function cmfConnect({
 				}
 			}
 
+			onEvent() {
+				return Object.keys(this.props).reduce(this.onEventDispatch, { toOmit: [] });
+			}
+
+			onEventDispatch(props, key) {
+				if (key.startsWith('on')) {
+					if (key.endsWith('Dispatch')) {
+						props.toOmit.push(key);
+						// eslint-disable-next-line no-param-reassign
+						props[key.replace('Dispatch', '')] = (event, data) => {
+							const payload = Object.assign(
+								{
+									event: serializeEvent(event),
+									data,
+								},
+								this.props[key],
+							);
+							this.props.dispatch(payload);
+						};
+					}
+					if (key.endsWith('ActionCreator')) {
+						props.toOmit.push(key);
+						const newKey = key.replace('ActionCreator', '');
+						let actionCreator = this.props[key];
+						let args;
+						if (typeof this.props[key] === 'object') {
+							actionCreator = this.props[key].id;
+							args = this.props[key].args;
+						}
+						// eslint-disable-next-line no-param-reassign
+						props[newKey] = (event, data) => {
+							if (!args) {
+								args = [
+									serializeEvent(event),
+									{
+										props: this.props,
+										data,
+									},
+								];
+							}
+							this.dispatchActionCreator(actionCreator, ...args);
+						};
+					}
+				}
+				return props;
+			}
+
 			dispatchActionCreator(actionCreatorId, event, data, context) {
 				const extendedContext = Object.assign({}, this.context, context);
 				this.props.dispatchActionCreator(actionCreatorId, event, data, extendedContext);
 			}
 
 			render() {
+				const { toOmit, ...handlers } = this.onEvent();
 				const props = {
-					...this.props,
+					...omit(this.props, toOmit),
+					...handlers,
```

Comment:
issue here, thanks @jare, we drop onEvent if it was on the props.
We must take care of composition in the build of the function onEvent